Lib/pdb.py

```diff
@@ -414,16 +416,34 @@ def preloop(self):
                     self.message('display %s: %r  [old: %r]' %
                                  (expr, newvalue, oldvalue))
 
-    def interaction(self, frame, traceback):
+    def interaction(self, frame, tb_or_exc):
         # Restore the previous signal handler at the Pdb prompt.
+
+        if isinstance(tb_or_exc, BaseException):
+            tb_or_exc, exception = tb_or_exc.__traceback__, tb_or_exc
+            self._chained_exceptions = []
+            self._chained_exception_index = -1
+            current = exception
+            while current:
+                self._chained_exception_index += 1
```

Comment:
Why do we need to maintain `_chained_exception_index` in this loop?

Reply:
You are right, we can just use the length after populating the list.